xbmc/application/ApplicationPlayerCallback.cpp

```diff
@@ -244,17 +221,8 @@ void CApplicationPlayerCallback::OnPlayBackSeek(int64_t iTime, int64_t seekOffse
                                                static_cast<int>(seekOffset));
 #endif
 
-  CVariant param;
-  JSONRPC::CJSONUtils::MillisecondsToTimeObject(iTime, param["player"]["time"]);
-  JSONRPC::CJSONUtils::MillisecondsToTimeObject(seekOffset, param["player"]["seekoffset"]);
-  param["player"]["playerid"] = CServiceBroker::GetPlaylistPlayer().GetCurrentPlaylist();
-  const auto& components = CServiceBroker::GetAppComponents();
-  const auto appPlayer = components.GetComponent<CApplicationPlayer>();
-  param["player"]["speed"] = static_cast<int>(appPlayer->GetPlaySpeed());
-  CServiceBroker::GetAnnouncementManager()->Announce(ANNOUNCEMENT::Player, "OnSeek",
-                                                     m_itemCurrentFile, param);
-
-  CDataCacheCore::GetInstance().SeekFinished(static_cast<int>(seekOffset));
+  CGUIMessage msg(GUI_MSG_PLAYBACK_SEEKED, 0, 0, iTime, seekOffset);
```

Comment:
I doubt this needs to be an int64_t, would a uint32_t be enough?

Reply:
Video player uses uin64_t for both values:
https://github.com/xbmc/xbmc/blob/d3f1d4b10afcff29d4cf49a8f44a49cb24710071/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3192C2-L3192C2
https://github.com/xbmc/xbmc/blob/d3f1d4b10afcff29d4cf49a8f44a49cb24710071/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3217C1-L3217C1

Reply:
I'm all for changing to int64_t, but this will cause a *lot* of warnings anywhere GetParamX() is currently used in a calculation and assigned to an int. A change to int64_t should probably come with the brush of a fine-toothed comb.

Reply:
@garbear I understand your concerns. Although casting each value returned from GetParamX() seems to be an excessive task to implement. m_param1 contains values from constant expressions like GUI_MSG_UPDATE_ITEM, ACTION_SELECT_ITEM. Changing their type to int64_t would introduce inconsistency across other message constants.

I've got two proposals. Please let me know if one of them is acceptable to you (to be honest I'm not a huge fan of them but I didn't come up with anything else).

1. 
```
int CGUIMessage::GetParam1() const
{
  return static_cast<int>(m_param1);
}

int64_t CGUIMessage::GetParam1i64() const
{
  return m_param1;
}

int CGUIMessage::GetParam2() const
{
  return static_cast<int>(m_param2);
}

int64_t CGUIMessage::GetParam2i64() const
{
  return m_param2;
}
```

Narrowing conversion happens in getters - there are no code changes needed from the caller perspective.

3. Leave int in m_paramX and use CGUIMessage::SetPointer/GetPointer to send POD like
```
struct
{
  const int64_t iTime;
  const int64_t seekOffset;
};
```

Reply:
Approach 1 looks better (though I would call it `GetParam1AsI64()`).

Approach 2 is certainly creative but let's not extend the use of raw pointers.

Overall the "param1/param2" think is a bit hacky. We could add "param3" and "param4" for 64bit ints but that just makes the problem worse.

I'm fine with Approach 1 but I'm open to other ideas that are less hacky.

Reply:
I've updated the PR. Now it implements Approach 1.